modules/install-consul/install-consul

```diff
@@ -166,8 +166,46 @@ function fetch_binary {
   local -r version="$1"
   local download_url="$2"
 
-  if [[ -z "$download_url" && -n "$version" ]];  then
-    download_url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
+  local -r cpu_arch="$(uname -m)"
```

Comment:
Declare variables and use subshells in separate statements, as otherwise, the exit code from the subshell will be lost, and we only get the exit code from `local`.

```suggestion
  local cpu_arch
  cpu_arch="$(uname -m)"
```

Reply:
Applied the suggestion 🥂